dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/hibernate/jsonb/type/JsonDeviceRenderTypeMap.java

```diff
@@ -28,38 +28,32 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JavaType;
 import org.hisp.dhis.render.DeviceRenderTypeMap;
+import org.hisp.dhis.render.RenderDevice;
+import org.hisp.dhis.render.type.RenderingObject;
 
 import java.io.IOException;
+import java.util.LinkedHashMap;
 import java.util.Properties;
 
 public class JsonDeviceRenderTypeMap extends JsonBinaryType
 {
-    private Class<? extends JsonDeserializer> deserializer;
+    private Class<? extends RenderingObject> renderType;
 
     @Override
-    protected Object convertJsonToObject( String content )
+    protected JavaType getResultingJavaType( Class<?> returnedClass )
     {
-        try
```

Comment:
I'm not very familiar with this renderer, but this look like a drastic change. Can you comment what has been done here?

Reply:
In an attempt to fix the issue, I have removed AbstractDeviceRenderTypeMapDeserializer which I have added to support XML deserialization of RenderType. So the content of the method `JsonDeviceRenderTypeMap.convertMapToObject()` is actually copied from the deserializer. Even if this doesn't solve the issue, I think it reduces the complexity of the code.
This is covered by the test `MetadataImportServiceTest.testSerializeDeviceRenderTypeMap()`